eras/conway/impl/testlib/Test/Cardano/Ledger/Conway/Imp/GovSpec.hs

```diff
@@ -1092,6 +1091,50 @@ spec =
         expectExtraDRepExpiry drep 2
 
     describe "Committee actions validation" $ do
+      it "A CC that has resigned will need to be first voted out and then voted in to be considered active" $ do
+        (dRep, _, gaidCC) <- electBasicCommittee
+        passNEpochs 2
+        -- Add a fresh CC
+        cc <- KeyHashObj <$> freshKeyHash
+        let addCCAction = UpdateCommittee (SJust gaidCC) mempty (Map.singleton cc 20) (1 %! 2)
+        addCCGaid <- submitGovAction addCCAction
+        submitYesVote_ (DRepVoter dRep) addCCGaid
+        passNEpochs 2
+        -- Confirm that they are added
+        SJust committee <- getsNES $ nesEsL . esLStateL . lsUTxOStateL . utxosGovStateL . committeeGovStateL
+        Map.keys (committee ^. committeeMembersL) `shouldContain` [cc]
+        -- Confirm their hot key registration
+        _hotKey <- registerCommitteeHotKey cc
+        ccShouldNotBeResigned cc
+        -- Have them resign
+        resignCommitteeColdKey cc
+        ccShouldBeResigned cc
+        -- Re-add the same CC
+        let reAddCCAction = UpdateCommittee (SJust $ GovPurposeId addCCGaid) mempty (Map.singleton cc 20) (1 %! 2)
+        reAddCCGaid <- submitGovAction reAddCCAction
+        submitYesVote_ (DRepVoter dRep) reAddCCGaid
+        passNEpochs 2
+        -- Confirm that they are still resigned
+        ccShouldBeResigned cc
+        -- Remove them
+        let removeCCAction = UpdateCommittee (SJust $ GovPurposeId reAddCCGaid) (Set.singleton cc) mempty (1 %! 2)
+        removeCCGaid <- submitGovAction removeCCAction
+        submitYesVote_ (DRepVoter dRep) removeCCGaid
+        passNEpochs 2
+        -- Confirm that they have been removed
+        SJust committee' <- getsNES $ nesEsL . esLStateL . lsUTxOStateL . utxosGovStateL . committeeGovStateL
+        Map.keys (committee' ^. committeeMembersL) `shouldNotContain` [cc]
+        -- Add the same CC back a second time
+        let secondAddCCAction = UpdateCommittee (SJust $ GovPurposeId removeCCGaid) mempty (Map.singleton cc 20) (1 %! 2)
+        secondAddCCGaid <- submitGovAction secondAddCCAction
+        submitYesVote_ (DRepVoter dRep) secondAddCCGaid
+        passNEpochs 2
+        -- Confirm that they have been added
+        SJust committeeAfterRemoveAndAdd <- getsNES $ nesEsL . esLStateL . lsUTxOStateL . utxosGovStateL . committeeGovStateL
+        Map.keys (committeeAfterRemoveAndAdd ^. committeeMembersL) `shouldContain` [cc]
```

Comment:
Avoid membership checks on lists at all costs, especially since shouldContain is O(n^2) complexity. I know this is just a unit test, but having a good habit of writing faster code at all times goes a long way.

Also it looks like a repeating pattern that happens 3 times in this test, so it might be a good idea to create a helper function for it.
```suggestion
        assertBool "Expected CC to be present in the committee" $ 
          Map.member cc (committeeAfterRemoveAndAdd ^. committeeMembersL) 
```

Reply:
Noted! Thanks @lehins 🙌